shipyard_blueprints/googledrive/shipyard_googledrive/drive_utils.py

```diff
@@ -191,20 +193,36 @@ def get_folder_id(
         if is_folder_id(folder_identifier):
             return folder_identifier
         else:
-            results = (
-                service.files()
-                .list(
-                    q=f"name = '{folder_identifier} and mimeType = 'application/vnd.google-apps.folder'",
-                    fields="files(id)",
+            if not drive_id:
+                results = service.files().list()
+            else:
+                results = (
+                    service.files()
+                    .list(
+                        q=f"name = '{folder_identifier}' and mimeType = 'application/vnd.google-apps.folder'",
+                        fields="files(id)",
+                        supportsAllDrives=True,
+                        includeItemsFromAllDrives=True,
+                        corpora="drive",
+                        driveId=drive_id,
+                    )
+                    .execute()
                 )
-                .execute()
-            )
             folders = results.get("files", [])
+            if len(folders) > 1:
+                raise ExitCodeException(
+                    f"Multiple folders with name {folder_identifier} found, please use the folder ID instead",
+                    204,
+                )
+
             if folders:
                 return folders[0]["id"]
             else:
                 return None
-    except Exception:
+
+    except ExitCodeException as ec:
+        raise (ExitCodeException(ec.message, ec.exit_code))
```

Comment:
Not preventing me from approving but a small stylistic suggestion. remove the extra `(` `)`